analyses/immune-deconv/01-immune-deconv.R

```diff
@@ -87,13 +91,11 @@ deconv <- function(expr_input, clin_file, method) {
   return(res)
 }
 
-# Deconvolute using xCell for poly-a and stranded datasets
+# Deconvolute using specified method for poly-a and stranded datasets
 expr_input <- c("polya", "stranded")
 combo <- expand.grid(expr_input, deconv_method, stringsAsFactors = F) 
 deconv_output <- apply(combo, 1, FUN = function(x) deconv(expr_input = x[1], clin_file = clin_file, method = x[2]))
 deconv_output <- do.call(rbind.data.frame, deconv_output)
 
-# save output to RData object
-print("Writing output to file..")
-save(deconv_output, file = output_file)
-print("Done!")
+# save output to RDS file
+write_rds(deconv_output, output_file)
```

Comment:
To check my understanding - the `deconvolute()` steps are run for each dataset separately, but then we're putting them into the same data frame. For xCell, the scores don't change depending on whether the data are together or not (see discussion on #292). I don't know that the same is true of `quantiseq`, but what I'm actually getting at is that our result almost implies that we ran them together when that's not the case. I think this (rambling) comment could be addressed by adding a column in the resulting data frame that keeps track of which input dataset the sample comes from.

Reply:
:100: